src/main.cc

```diff
@@ -1927,6 +1939,24 @@ Parse(TRITONSERVER_ServerOptions** server_options, int argc, char** argv)
             std::get<2>(bcs).c_str()),
         "setting backend configurtion");
   }
+  for (const auto& limit : load_gpu_limit) {
+    if (limit.first < 0) {
+      std::cerr << "--model-load-gpu-limit option expects device ID >= 0, Got "
+                << limit.first << std::endl;
+      exit(1);
+    } else if ((limit.second < 0.0) || (limit.second > 1.0)) {
+      std::cerr << "--model-load-gpu-limit option expects limit fraction to be "
+                   "in range [0.0, 1.0], Got "
+                << limit.second << std::endl;
+      exit(1);
+    }
+    static std::string key_prefix = "model-load-gpu-limit-device-";
+    FAIL_IF_ERR(
+        TRITONSERVER_ServerOptionsSetBackendConfig(
```

Comment:
Why are we using the `TRITONSERVER_ServerOptionsSetBackendConfig` field? This looks like a ServerOptions and not a backend field.

Reply:
This may not the intended usage as we didn't document it, but config for "empty backend" are used to pass in global configs which will be accessed by model related components (i.e. we pass `backend-directory` this way). It is probably more consistent if I add new API in a similar way to how we set backend directory